### PR TITLE
[continue:] Moving towards multi-sensor-per-same-measurement fusion

### PR DESCRIPTION
**Pull request by kamilritz**
This PR is exactly the same as: https://github.com/PX4/ecl/pull/652,
but added as a separate branch to ECL. Such that we can test the PR's compatibility with the Firmware's CI.

### CODE REVIEW

**Comment by mhkabir** [comment]
Question: why do we even use the std. deviation in the estimator interface?
1. PX4 side uses variances (from cov matx)
2. Fusion uses variances

So we just end up uselessly square rooting and then squaring.

File: EKF/vel_pos_fusion.cpp [hunk 1]
```diff
@@ -45,276 +45,174 @@
...
+		_time_last_ver_vel_fuse = _time_last_imu;
+		_innov_check_fail_status.flags.reject_ver_vel = false;
 
-		// Set observation noise variance and innovation consistency check gate size for the NE position observations
-		R[4] = R[3] = sq(_posObsNoiseNE);
-		gate_size[4] = gate_size[3] = _posInnovGateNE;
+		fuseVelPosHeight(innov(2),innov_var(2),2);
 
+		return true;
+	}else{
```
Comment (mhkabir) [inline EKF/vel_pos_fusion.cpp]:
```suggestion
	} else {
```

**Comment by kamilritz** [comment]
@mhkabir I also noticed that and want get rid of the std deviation. I am in the process of cleaning this and some other things up. But these changes are based on top of this refactor. I am eager to get them in after this PR.

File: EKF/vel_pos_fusion.cpp (hunk 1 above)
Comment (kamilritz) [reply EKF/vel_pos_fusion.cpp]: That is only the tip of the iceberg. At some point we really need a formatter.

**Comment by mhkabir** [comment]
@kamilritz What is missing here? Let's bring this in soon.

**Comment by kamilritz** [comment]
@mhkabir Glad you feel the same. Waiting on the firmware PR to be ready.

**Comment by xdwgood** [comment]
@kamilritz  https://github.com/PX4/ecl/blob/master/EKF/vel_pos_fusion.cpp#L168

This is just `KP`, why does the code say it is `KHP`, where does `H` come from?  thank you very much :smiley:

**Comment by kamilritz** [comment]
> Is it because the H matrix is 1?

Yes. To be a bit more precise. It is a vector of size 1x24 full of zeros with a single 1 at the location of the measurement.

**Pull request merged**